src/components/Header.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { ChangeEvent, FormEvent } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  newTodoTitle: string,
+  setNewTodoTitle: (title: string) => void,
+  addTodo: (event: FormEvent<HTMLFormElement>) => void,
+  isLoading: boolean,
+  toggleAllcompletedTodos: () => void,
+  todos: Todo[],
+};
+
+export const Header: React.FC<Props> = ({
+  newTodoTitle,
+  setNewTodoTitle,
+  addTodo,
+  isLoading,
+  toggleAllcompletedTodos,
+  todos,
+}) => {
+  const handleAddTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const allCompletedTodos = todos.every(todo => todo.completed);
```

Comment:
Consider adding a check for empty 'todos' array before using 'every()' to avoid marking 'allCompletedTodos' as true when there are no todos.